### DISCUSSION TITLE
Discussion about the need for the @celo/rainbowkit-celo config in its current form

### DISCUSSION THREAD

**Discussion by arthurgousset** [celo-org/rainbowkit-celo#85] [General]
## TLDR

Discussing the need for the [`@celo/rainbowkit-celo`](https://www.npmjs.com/package/@celo/rainbowkit-celo) NPM package in its current form, considering:

- the value it provides to the Celo developer experience,
- the ongoing maintenance cost it represents (relative to our other maintenance commitments), and
- how we can provide a similar developer experience without the maintenance cost.

## Context 

This discussion was originally started here:

- https://github.com/celo-org/rainbowkit-celo/issues/84

> Wagmi + Viem v2 are live. Rainbowkit has published a v2 beta to match it.
> 
> Relevant docs: https://beta.rainbowkit.com/guides/rainbowkit-wagmi-v2 https://viem.sh/docs/migration-guide#2xx-breaking-changes https://wagmi.sh/react/guides/migrate-from-v1-to-v2

> Alternatively, it may be better to just get the wallets added to rainbowkit's official list: https://github.com/rainbow-me/rainbowkit/tree/main/packages/rainbowkit/src/wallets/walletConnectors
> 
> Viem already has celo in its chain consts so the need for this package is diminished

> Seems to work: https://github.com/jmrossy/celo-station/blob/wagmi-v2/src/config/wallets.ts

> alternatively consider deprecating package as (as @jmrossy mentioned) the need for it has diminished and there has proved to be high maintenance costs for such a small config.

**Comment by arthurgousset** [comment]
If I understand correctly, we can provide a similar developer experience if we add Celo-specific wallets to rainbowkit's ([`@rainbow-me/rainbowkit`](https://www.npmjs.com/package/@rainbow-me/rainbowkit)) official wallet list ([walletConnectors](https://github.com/rainbow-me/rainbowkit/tree/main/packages/rainbowkit/src/wallets/walletConnectors)).

As mentioned, [`viem`](https://www.npmjs.com/package/viem) already publishes Celo chain configs for mainnet ([`celo.ts`](https://github.com/wevm/viem/blob/main/src/chains/definitions/celo.ts)) and alfajores testnet ([`celoAlfajores.ts`](https://github.com/wevm/viem/blob/main/src/chains/definitions/celoAlfajores.ts)).

Seems like the next step is to make a PR to add Celo-specific wallets to rainbowkit, and see if it gets merged.

**Reply by jmrossy** [reply]
Agreed

**Reply by arthurgousset** [reply]
GitHub issue to track the status: 
- #86

**Reply by arthurgousset** [reply]
cc @celo-org/devtooling for visibility
